PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsDotNetCoreIsolatedDebugProfile.kt

```diff
@@ -0,0 +1,169 @@
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jetbrains.plugins.azure.functions.run
+
+import com.intellij.execution.DefaultExecutionResult
+import com.intellij.execution.ExecutionResult
+import com.intellij.execution.Executor
+import com.intellij.execution.process.ProcessAdapter
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.execution.runners.ProgramRunner
+import com.intellij.execution.ui.ConsoleView
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.util.Key
+import com.intellij.util.execution.ParametersListUtil
+import com.jetbrains.rd.util.lifetime.Lifetime
+import com.jetbrains.rdclient.util.idea.pumpMessages
+import com.jetbrains.rider.debugger.DebuggerHelperHost
+import com.jetbrains.rider.debugger.DebuggerWorkerPlatform
+import com.jetbrains.rider.debugger.DebuggerWorkerProcessHandler
+import com.jetbrains.rider.model.debuggerWorker.DebuggerStartInfoBase
+import com.jetbrains.rider.model.debuggerWorker.DotNetCoreAttachStartInfo
+import com.jetbrains.rider.run.*
+import com.jetbrains.rider.runtime.DotNetExecutable
+import com.microsoft.intellij.util.PluginUtil
+import org.apache.commons.lang.StringUtils
+import org.jetbrains.plugins.azure.RiderAzureBundle
+import java.time.Duration
+
+class AzureFunctionsDotNetCoreIsolatedDebugProfile(
+        private val dotNetExecutable: DotNetExecutable,
+        executionEnvironment: ExecutionEnvironment)
+    : DebugProfileStateBase(executionEnvironment) {
+
+    companion object {
+        private val logger = Logger.getInstance(AzureFunctionsDotNetCoreIsolatedDebugProfile::class.java)
+        private const val DOTNET_ISOLATED_DEBUG_ARGUMENT = "--dotnet-isolated-debug"
+    }
+
+    private var processId = 0
+    private lateinit var targetProcessHandler: ProcessHandler
+    private lateinit var console: ConsoleView
+
+    override suspend fun createWorkerRunInfo(lifetime: Lifetime, helper: DebuggerHelperHost, port: Int): WorkerRunInfo {
+        // Launch Azure Functions host process
+        launchAzureFunctionsHost()
+
+        // Wait until we get a process ID (or the process terminates)
+        pumpMessages(Duration.ofMinutes(5)) {
+            processId != 0 || targetProcessHandler.isProcessTerminated
+        }
+        if (processId == 0) {
+            logger.error("Azure Functions host did not return isolated worker process id.")
+
+            // REVIEW: If we do not get pid from the isolated worker process, should we destroy the process here?
+            if (!targetProcessHandler.isProcessTerminating && !targetProcessHandler.isProcessTerminated) {
+                targetProcessHandler.destroyProcess()
+            }
+
+            // REVIEW: Should we throw here instead?
```

Comment:
I think we need to do both (show notification and throw). No need to proceed with PID = 0